src/subscription_manager/managercli.py

```diff
@@ -1452,6 +1452,13 @@ def _do_command(self):
             # TODO: grab the list of valid options, and check
             self.cp.updateConsumer(consumer['uuid'], release=self.options.release)
 
+        # Sync the syspurpose cache
+        syspurpose_dict = {'service_level_agreement': consumer['serviceLevel'],
+                          'role': consumer['role'],
+                          'usage': consumer['usage'],
+                          'addons': consumer['addOns']}
```

Comment:
I would try to get instance of `SyncedStore` using `_get_synced_store()`. If this method would not return `None`, then I would call the `update_cache()`.